src/System.Text.Json/docs/writable_json_dom_spec.md

```diff
@@ -0,0 +1,253 @@
+    {
+        ""name"" : ""Zoe"",
+        ""surname"" : ""Coder"",
+        ""age"" : 24,                
+    }
+}";
+
+JsonObject employees = JsonNode.Parse(jsonString) as JsonObject;
+employees.Add(EmployeesDatabase.GetNextEmployee());
+Mailbox.SendAllEmployeesData(employees.AsJsonElement());
+```
+
+## Design choices
+
+* Avoid any significant perf regression to the readonly implementation of `JsonDocument` and `JsonElement`.
+* Higher emphasis on usability over allocations/performance.
+* No advanced methods for looking up properties like `GetAllValuesByPropertyName` or `GetAllPrimaryTypedValues`, because they would be too specialized.
+* Support for LINQ style quering capability.
+* `null` reference to node instead of `JsonNull` class.
+
+* Initializing JsonArray with additional constructors accepting `IEnumerable`s of all primary types (bool, string, int, double,  long...).
+
+    Considered solutions:
+
+    1. One additional constructor in JsonArray
+    ```csharp
+    public JsonArray(IEnumerable<object> jsonValues) { }
+    ``` 
+    2. Implicit operator from Array in JsonArray
+
+    3. More additional constructors in JsonArray (chosen)
+    ```csharp
+    public JsonArray(IEnumerable<string> jsonValues) { }
+    public JsonArray(IEnumerable<bool> jsonValues) { }
+    public JsonArray(IEnumerable<sbyte> jsonValues) { }
+    ...
+    public JsonArray(IEnumerable<double> jsonValues) { }
+    ``` 
+
+    | Solution | Pros | Cons | Comment |
+    |----------|:-------------|:------|--------:|
+    | 1 | - only one additional method <br> - accepts collection of different types <br> - accepts `IEnumerable` <br> - IntelliSense (autocompletion and showing suggestions) | - accepts collection of types not deriving from `JsonNode` <br> - needs to check it in runtime  | accepts too much, <br> array of different primary types wouldn't be returned from method |
+    | 2 | - only one additional method <br> - accepts collection of different types <br > | - works  only in C# <br> - no IntelliSense <br> - users may not be aware of it <br> - accepts only `Array` <br> - accepts collection of types not deriving from `JsonNode` <br> - needs to check it in runtime | from {1,2}, <br>2 seems worse |
+    | 3 | - accepts IEnumerable <br> - does not accept collection of  types not deriving from `JsonNode` <br> - no checks in runtime <br> - IntelliSense | - a lot of additional methods <br> - does not accept a collection of different types | gives less possibilities than {1,2}, but requiers no additional checks |
+
+* Implicit operators for `JsonString`, `JsonBoolean` and `JsonNumber` as an additional feature.
+* `Sort` not implemented for `JsonArray`, beacuse there is no right way to compare `JsonObject`s. If user wants to sort `JsonArray` of `JsonNumber`s, `JsonBooleans`s or `JsonStrings` he/she now needs to do the following: convert `JsonArray` to regular array (by iterating through all elements), calling sort (and converting back to `JsonArray` if needed).
```

Comment:
nit: for consistency
```suggestion
* `Sort` not implemented for `JsonArray`, beacuse there is no right way to compare `JsonObject`s. If user wants to sort `JsonArray` of `JsonNumber`s, `JsonBooleans`s or `JsonString`s he/she now needs to do the following: convert `JsonArray` to regular array (by iterating through all elements), calling sort (and converting back to `JsonArray` if needed).
```

Although, having it formatted with the `s` outside the `<code block>` makes it difficult to read/parse the sentence.

Including other feedback
```suggestion
* `Sort` not implemented for `JsonArray`, beacuse there is no right way to compare `JsonObject`s. If a user wants to sort a `JsonArray` of `JsonNumber`s, `JsonBooleans`s or `JsonStrings` they need to do the following: convert the `JsonArray` to a regular array (by iterating through all elements), call sort (and convert back to `JsonArray` if needed).
```

Reply:
I can change everything to ...s`